F824 (Flake8) violation in Bio.PDB.vectors

Dear maintainers,

# Issue

Recently I *attempted* to submit a PR related to a small fix in the `vectors()` function within `Bio.PDB.vectors` (https://github.com/biopython/biopython/pull/5045), and although the code I submitted passed the checks, the PR failed the CI / style check with the following errors, which point to code already in the `main` branch:

```python3
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

Bio/PDB/vectors.py:551:5: F824 `global gtm` is unused: name is never assigned in scope
Bio/PDB/vectors.py:552:5: F824 `global gmry` is unused: name is never assigned in scope
Bio/PDB/vectors.py:553:5: F824 `global gmrz` is unused: name is never assigned in scope
Bio/PDB/vectors.py:553:5: F824 `global gmrz2` is unused: name is never assigned in scope
```

These F824 complaints refer to this part of the code (which, for clarity, is unrelated to my PR)

```python3
    global gtm
    global gmry
    global gmrz, gmrz2

    tm = gtm
    mry = gmry
    mrz = gmrz
    mrz2 = gmrz2

    # tx acs[1] to origin
    # tm = homog_trans_mtx(-a1[0][0], -a1[1][0], -a1[2][0])
    set_homog_trans_mtx(-a1[0], -a1[1], -a1[2], tm)
```

Where indeed, instead of doing something like `gmry=homog_trans_mtx(...)`, the `homog_trans_mtx()` function (and some similar ones) directly changes the values:

```python3
def set_homog_trans_mtx(x: float, y: float, z: float, mtx: np.ndarray):
    """Update existing translation matrix to new values."""
    mtx[0][3] = x
    mtx[1][3] = y
    mtx[2][3] = z
```

# Significance

The F824 violation is not a bug or an issue *per se*, but makes certain style checks fail, which virtually makes it impossible to pass checks on PRs that touch code within `Bio.PDB.vectors`

# Proposed solution

In my mind, and please correct me if I am wrong, this issue could be solved by refactoring the code, to something like

```python
def set_homog_trans_mtx(x: float, y: float, z: float, mtx: np.ndarray) -> mtx:
    """Update existing translation matrix to new values."""
    mtx[0][3] = x
    mtx[1][3] = y
    mtx[2][3] = z

    return mtx

mtx = set_homo_trans_mtx(x, y, z, mtx)
```

Thoughts on this? If the maintainers think this is a good approach, I would be happy to contribute to this.

--- a/Bio/PDB/vectors.py
+++ b/Bio/PDB/vectors.py
@@ -548,11 +548,6 @@ def coord_space(
     # a1 = acs[1]
     # a2 = acs[2]
 
-    # these global statements incur into an F824 (flake8) complaint
-    # global gtm
-    # global gmry
-    # global gmrz, gmrz2
-
     # the gtm, gmry, gmrz, and gmrz2 variables are read from the global namespace
     tm = gtm
     mry = gmry

--- a/Bio/PDB/vectors.py
+++ b/Bio/PDB/vectors.py
@@ -94,7 +94,6 @@ def vector_to_axis(line, point):
     """
     line = line.normalized()
     np = point.norm()
-    print("Test")
     angle = line.angle(point)
     return point - line ** (np * np.cos(angle))
 

--- a/Bio/PDB/vectors.py
+++ b/Bio/PDB/vectors.py
@@ -549,10 +549,12 @@ def coord_space(
     # a1 = acs[1]
     # a2 = acs[2]
 
+    # these global statements incur into an F824 (flake8) complaint
     # global gtm
     # global gmry
     # global gmrz, gmrz2
 
+    # the gtm, gmry, gmrz, and gmrz2 variables are read from the global namespace
     tm = gtm
     mry = gmry
     mrz = gmrz

--- a/Bio/PDB/vectors.py
+++ b/Bio/PDB/vectors.py
@@ -94,6 +94,7 @@ def vector_to_axis(line, point):
     """
     line = line.normalized()
     np = point.norm()
+    print("Test")
     angle = line.angle(point)
     return point - line ** (np * np.cos(angle))
 
@@ -548,9 +549,9 @@ def coord_space(
     # a1 = acs[1]
     # a2 = acs[2]
 
-    global gtm
-    global gmry
-    global gmrz, gmrz2
+    # global gtm
+    # global gmry
+    # global gmrz, gmrz2
 
     tm = gtm
     mry = gmry

--- a/CONTRIB.rst
+++ b/CONTRIB.rst
@@ -259,6 +259,7 @@ please open an issue on GitHub or mention it on the mailing list.
 - Micky Yun Chan <https://github.com/michiboo>
 - Mike Moritz <https://github.com/mikepmikepm35>
 - Mike Poidinger <Michael.Poidinger at domain eBioinformatics.com>
+- Mikel Martinez Goikoetxea <https://github.com/Mikel-MG>
 - Milind Luthra <https://github.com/milindl>
 - morrme <https://github.com/morrme>
 - Mustafa Anil Tuncel <https://github.com/anilbey>

